mediacheck.h

```diff
@@ -0,0 +1,60 @@
+#ifndef _MEDIACHECK_H
+#define _MEDIACHECK_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct mediacheck_digest_s mediacheck_digest_t;
+
+typedef int (* mediacheck_progress_t)(unsigned percent);
+
+typedef struct {
+  char *file_name;				/* file to check */
+  mediacheck_progress_t progress;		/* progress function */
+
+  unsigned full_blocks;				/* full image size, in 0.5 kB units */
+  unsigned iso_blocks;				/* iso size in 0.5 kB units */
+  unsigned pad_blocks;				/* padding size in 0.5 kB units */
+  unsigned part_start;				/* partition start, in 0.5 kB units */
+  unsigned part_blocks;				/* partition size, in 0.5 kB units */
+
+  struct {
+    mediacheck_digest_t *full;			/* full image digest, calculated */
+    mediacheck_digest_t *iso;			/* iso digest, calculated */
+    mediacheck_digest_t *part;			/* partition digest, calculated */
+  } digest;
+
+  struct {
+    char *key, *value;
+  } tags[16];					/* up to 16 key - value pairs */
+
+  unsigned abort:1;				/* check aborted */
+  unsigned err:1;				/* read error */
+  unsigned err_block;				/* read error position (in 0.5 kB units) */
+
+  char app_id[0x81];				/* application id */
+  char app_data[0x201];				/* app specific data*/
+
+  int last_percent;				/* last percentage shown by progress function */
+} mediacheck_t;
+
+mediacheck_t *mediacheck_init(char *file_name, mediacheck_progress_t progress);
+void mediacheck_done(mediacheck_t *media);
+void mediacheck_calculate_digest(mediacheck_t *media);
+
+mediacheck_digest_t *mediacheck_digest_init(char *digest_name, char *digest_value);
+void mediacheck_digest_done(mediacheck_digest_t *digest);
+void mediacheck_digest_process(mediacheck_digest_t *digest, unsigned char *buffer, unsigned len);
+
+int mediacheck_digest_valid(mediacheck_digest_t *digest);
+int mediacheck_digest_ok(mediacheck_digest_t *digest);
+char *mediacheck_digest_name(mediacheck_digest_t *digest);
+char *mediacheck_digest_hex(mediacheck_digest_t *digest);
+char *mediacheck_digest_hex_ref(mediacheck_digest_t *digest);
```

Comment:
well, header should be source of documentation for library. For some methods I have no clue what it did. Is possible to document them? At least shortly.

Reply:
I know it is documented in code. But I think in C and C++ it is more common to document public methods in header file which people read when reading about API to use when using library.